plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -487,6 +487,8 @@ compileExpr e = withContextM 2 (sdToTxt $ "Compiling expr:" GHC.<+> GHC.ppr e) $
                     GHC.$+$ (GHC.ppr $ GHC.realIdUnfolding n)
 
         -- arg can be a type here, in which case it's a type instantiation
+        -- we ignore types of 'RuntimeRep' kind to compile unboxed tuples properly
```

Comment:
too vague: I think you want to write a note about all this stuff and refer to it here. You've figured out a lot of things: let's write them down!